tarpc/src/rpc/client/channel.rs

```diff
@@ -106,7 +115,7 @@ impl<Req, Resp> Channel<Req, Resp> {
 
         let (response_completion, response) = oneshot::channel();
         let cancellation = self.cancellation.clone();
-        let request_id = self.next_request_id.fetch_add(1, Ordering::Relaxed);
+        let request_id = self.next_request_id.fetch_add(1, Ordering::Relaxed) as u64;
```

Comment:
```suggestion
        let request_id = u64.try_from(self.next_request_id.fetch_add(1, Ordering::Relaxed)).unwrap();
```

Even though there's already a compile-time check, in the spirit of defense in depth, I'd like to explicitly check that the conversion is sound.

Reply:
Additionally, looking at the implementation of try_from for targets with usize <= 64, they return `X as u64` unconditionally.
So in theory compiler optimizations should get rid of both try_from and unwrap anyway which means the code would be just as fast as before :)

Reply:
for reference: https://github.com/rust-lang/rust/blob/1.49.0/library/core/src/convert/num.rs